migrations/settings/0002-migrate-icons-setting.js

```diff
@@ -2,29 +2,25 @@ export default function migrate(settings) {
   const oldSetting = settings.get("icons");
 
   if (oldSetting) {
-    const newSetting = oldSetting.split("|").map((link) => {
-      let [iconName, url, target] = link.split(",").map((s) => s.trim());
-
-      if (["_blank", "_self", "_parent", "_top"].indexOf(target) === -1) {
-        target = "_blank";
-      }
+    const newIcons = [];
 
-      const newLink = {
-        icon_name: iconName,
-        url,
-        target
-      }
+    oldSetting.split("|").map((link) => {
+      let [iconName, url, target] = link.split(",").map((s) => s.trim());
 
-      Object.keys(newLink).forEach((key) => {
-        if (newLink[key] === undefined) {
-          delete newLink[key];
+      if (iconName && url) {
+        if (["_blank", "_self", "_parent", "_top"].indexOf(target) === -1) {
+          target = "_blank";
         }
-      });
 
-      return newLink;
-    })
+        newIcons.push({
+          icon_name: iconName,
+          url,
+          target,
+        });
+      }
+    });
 
-    settings.set("icons", newSetting);
+    settings.set("icons", newIcons);
```

Comment:
Jut to check, 
```
{
  icon_name: iconName,
  url: undefined,
  target: undefined,
}
```

is valid for settings? since the `split(",")` may land up with just one item in the array. (same for newLinks)

Reply:
We don't get undefined anymore since this is what the PR is trying to solve now. `target` will always be set and we only add a link if `icon_name` and `url` is not `undefined`.